vm/src/stdlib/sys.rs

```diff
@@ -442,6 +442,52 @@ mod sys {
         .into_struct_sequence(vm))
     }
 
+    #[pyfunction]
+    fn __unraisablehook__(
+        unraisable: PyRef<UnraisableHookArgs>,
+        vm: &VirtualMachine,
+    ) -> PyResult<()> {
+        use super::PyStderr;
+
+        let stderr = PyStderr(vm);
+        if !vm.is_none(&unraisable.object) {
+            if !vm.is_none(&unraisable.err_msg) {
+                write!(stderr, "{}: ", unraisable.err_msg.repr(vm)?);
+            } else {
+                write!(stderr, "Exception ignored in: ");
+            }
+            // exception in del will be ignored but printed
+            let repr = &unraisable.object.repr(vm);
+            let str = match repr {
+                Ok(v) => format!("{v}"),
+                Err(_) => format!(
+                    "<object {} repr() failed>",
+                    unraisable.object.class().name()
+                ),
+            };
+            writeln!(stderr, "{str}");
+        } else if !vm.is_none(&unraisable.err_msg) {
+            writeln!(stderr, "{}:", unraisable.err_msg.repr(vm)?);
```

Comment:
In cpython, use `__str__` only for this line of code. 
https://github.com/python/cpython/blob/main/Python/errors.c#L1316-L1323
```c
else if (err_msg != NULL && err_msg != Py_None) {
    if (PyFile_WriteObject(err_msg, file, Py_PRINT_RAW) < 0) {
        return -1;
    }
    if (PyFile_WriteString(":\n", file) < 0) {
        return -1;
    }
}
```
https://github.com/python/cpython/blob/main/Objects/fileobject.c#L128-L130
```c
if (flags & Py_PRINT_RAW) {
    value = PyObject_Str(v);
}
else
    value = PyObject_Repr(v);
```